### DISCUSSION TITLE
Why is a cylinder not a closed mesh?

### DISCUSSION THREAD

**Discussion by shakasaki** [pyvista/pyvista#4212] [Q&A] [answered]
Hello, I am trying to pick points that fall within a cylinder in the same way as they can be [picked in a sphere](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.select_enclosed_points.html), but only to realize that a cylinder is not a closed manifold in pyvista. Is there a way to get a closed cylinder to do this? I also tried CylinderStructured but it still does not work. Thanks for the help!

`import pyvista
cylinder = pyvista.Cylinder()
plane = pyvista.Plane()
selected = plane.select_enclosed_points(cylinder)
pts = plane.extract_points(selected['SelectedPoints'].view(bool),
                           adjacent_cells=False)
pl = pyvista.Plotter()
_ = pl.add_mesh(cylinder, style='wireframe')
_ = pl.add_points(pts, color='r')
pl.show()`

error :

`RuntimeError: Surface is not closed. Please read the warning in the documentation for this function and either pass `check_surface=False` or repair the surface.
`

**Comment by darikg** [comment, marked as answer]
it's not ideal, but it's easy to fix: just change `pyvista.Cylinder()` to `pyvista.Cylinder().clean()`. I think vtkCylinderSource creates the walls and caps separately and doesn't automatically merge the duplicated points that exist in common, so there's non-manifold topology until the clean filter merges them.

**Comment by shakasaki** [comment]
Excellent, works as expected, thanks!


```
import pyvista
cylinder = pyvista.Cylinder().clean()
plane = pyvista.Plane()
selected = plane.select_enclosed_points(cylinder)
pts = plane.extract_points(selected['SelectedPoints'].view(bool), adjacent_cells=False)
pl = pyvista.Plotter()
_ = pl.add_mesh(cylinder, style='wireframe')
_ = pl.add_points(pts, color='r')
pl.show()
```



![image](https://user-images.githubusercontent.com/7264182/229313329-fb9124e2-24e9-415b-9e82-8b71a7b35c91.png)
